lib/platform/src/linux/linux_x11clipboard.c

```diff
@@ -82,7 +82,7 @@ void ExecuteCallback(void)
 		return;
 	}
 	size_t len = clipboard.text_len;
-	wchar_t *wstr = malloc(sizeof(wchar_t) * len);
+	wchar_t *wstr = malloc(sizeof(wchar_t) * (len + 1));
```

Comment:
Is there a need for the `+1` here? I'm not quite sure if I'm doing the copying right. Would it be possible to extract this to a macro or a function, so these mistakes can be avoided?
@lc-soft

Reply:
@WhoAteDaCake  Yes. the string terminator `'\0'` also takes up memory, so we need `len + 1`.

Reply:
When decoding a string, we need to get the length of the decoded string before allocating memory.

```c
size_t len = decode_utf8(NULL, clipboard.text, 0);
wchar_t *wstr = malloc(sizeof(wchar_t) * (len + 1));
decode_utf8(wstr, clipboard.text, len);
wstr[len] = 0;
```